team/bundles/org.eclipse.team.ui/src/org/eclipse/team/internal/ui/history/LocalHistoryPage.java

```diff
@@ -567,8 +567,19 @@ protected void fillTableMenu(IMenuManager manager) {
 					if (sel instanceof IStructuredSelection) {
 						IStructuredSelection tempSelection = (IStructuredSelection) sel;
 						if (tempSelection.size() == 1) {
-							manager.add(new Separator("getContents")); //$NON-NLS-1$
-							manager.add(getContentsAction);
+							boolean showGetContentsAction = true;
```

Comment:
If only we were using a more recent Java, this would be a 1-liner.

```java
if (!(tempSelection.getFirstElement() instanceof LocalFileRevision lfr && lfr.isCurrentState())) {
  managed.add(...);
}
```

cc @akurtakov , in case you need some example convincing people that Java 17 would really make Ecipse code better.

Reply:
Less "lines" (you can always write this code already as one line) does not makes better code. While it seems "smarter" it makes the code hard to debug and understand.

For this particular case, a method that returns a boolean like 

```
if (canGetContent(sel)) { 
   managed.add(...); 
}
```

is (independent of the java version) much more readable, better understandable and easier to debug even though it might result in even more lines.

Reply:
And just for the record, the code can already be simplified as since 3.11 JFaces offers a 

`org.eclipse.jface.viewers.AbstractTreeViewer.getStructuredSelection()`

so cast in L567 can be avoided.

Reply:
I actually was wondering how far I can take the code to modern style ; I would have even used a var!  
Will trim things down a bit. @laeubi do you think this warrants a method extraction? Only use, can't even predict another usage...